frontend/src/views/consumer/app-modal/index.js

```diff
@@ -0,0 +1,102 @@
+import React, { Component } from 'react';
+import Modal, { Content } from '../../../components/modal';
+import Typography from '../../../components/typography';
+import { connect } from '../../../utils';
+import { withTheme } from 'styled-components';
+import Icon from 'antd/lib/icon';
+
+export default withTheme(
+  connect('store')(
+    class AppModal extends Component {
+      state = {
+        countDown: 5,
+      };
+      componentDidUpdate() {
+        // if the signal is handled, do nothing
+        if (this.state.redirectingLocally) return;
+        // when there is a redirecting signal from state, this components initiate a countdown
+        // when the countdown hits 0, it will redirect
+        if (this.props.store.ui.orderAndPayment.redirecting) {
+          console.log('Handling redirect signal');
+          this.setState(
+            {
+              countDown: 5,
+              redirectingLocally: true, // mark that the signal has been handled.
+            },
+            () => {
+              this.interval = window.setInterval(() => {
+                this.setState({
+                  countDown: this.state.countDown - 1,
+                });
+              }, 1000);
+            },
+          );
+        }
+      }
+      componentWillUnmount() {
+        window.clearInterval(this.interval);
+      }
+      render() {
+        const { orderAndPayment } = this.props.store.ui;
+        const { palette } = this.props.theme;
+
+        if (this.state.countDown < 1) {
```

Comment:
I think the render method is not the right place to do this... could you do it inside the setInterval callback?

Reply:
yes, i can move it to a componentDidUpdate lifecycle

Reply:
👍